charts/pmm-ha/templates/haproxy-init-configmap.yaml

```
@@ -20,15 +20,20 @@ data:
       
       echo "Checking PMM instance: $pmm_host:$pmm_port"
       
-      if curl -s --max-time "$timeout" --fail "http://$pmm_host:$pmm_port/v1/readyz" > /dev/null 2>&1; then
-        echo "✅ HTTP health check passed for $pmm_host:$pmm_port"
-        return 0
-      elif curl -s --max-time "$timeout" --fail -k "https://$pmm_host:$pmm_port/v1/readyz" > /dev/null 2>&1; then
-        echo "✅ HTTPS health check passed for $pmm_host:$pmm_port"
+      local http_code=$(curl -s --max-time "$timeout" -o /dev/null -w "%{http_code}" "http://$pmm_host:$pmm_port/")
```

```suggestion
      local http_code=$(curl -s --max-time "$timeout" -o /dev/null -k -w "%{http_code}" "https://$pmm_host:$pmm_port/")
```
Let's not promote http (aka insecure) calls. Also, skipping to check the certificates seems to fine for a readyz probe.